modules/alloy/src/main/ts/ephox/alloy/behaviour/coupling/CouplingState.ts

```diff
@@ -1,16 +1,27 @@
-import { Fun, Obj } from '@ephox/katamari';
+import { Fun, Obj, Optional } from '@ephox/katamari';
 
 import { AlloyComponent } from '../../api/component/ComponentApi';
 import { nuState } from '../common/BehaviourState';
 import { CouplingConfig, CouplingState } from './CouplingTypes';
 
+// Unfortunately, the Coupling APIs currently throw errors when the coupled name
+// is not recognised. This is because that if the wrong name is used, it is a
+// non-recoverable error, and the developer should be notified. However, there are
+// better ways to do this.
 const init = (): CouplingState => {
   const coupled: Record<string, AlloyComponent> = { };
 
   const getOrCreate = (component: AlloyComponent, coupleConfig: CouplingConfig, name: string): AlloyComponent => {
+    // These are the situations:
+    // 1. If we don't have any coupling configuration of any kind, then Error
```

Comment:
See previous comment about comments. There is arguably also value in an outdated comment, because it shows the assumptions that held when the code was written.

Part of these comments are designed to help someone identify if the code is wrong when they encounter it, by seeing if it follows the overall plan. For example, this code was wrong, but if someone just looked at the code, they may never have identified that, as they didn't know the intention. But I'm happy to take these comments out if that's the preferred option.

- [ ] remove comments.

Reply:
Updated: comment removed